eserde/src/toml.rs

```diff
@@ -0,0 +1,68 @@
+//! Deserialize TOML documents.
+
+use crate::{
+    path, reporter::ErrorReporter, DeserializationError, DeserializationErrors, EDeserialize,
+};
+use toml;
+use serde::Deserialize;
+
+/// Deserialize an instance of type `T` from a string of TOML text.
+///
+/// # Example
+///
+/// ```rust
+/// #[derive(eserde::Deserialize, Debug)]
+/// struct User {
+///     fingerprint: String,
+///     location: String,
+/// }
+///
+/// # fn main() {
+/// let data = r#"
+///     fingerprint = "1234567890ABCDEF"
+///     location = "Menlo Park, CA"
+/// "#;
+///
+/// let u: User = eserde::toml::from_str(data).unwrap();
+/// println!("{:#?}", u);
+/// # }
+/// ```
+pub fn from_str<'a, T>(s: &'a str) -> Result<T, DeserializationErrors>
+where
+    T: EDeserialize<'a>,
```

Comment:
```suggestion
pub fn from_str<T>(s: &str) -> Result<T, DeserializationErrors>
where
    T: for<'a> EDeserialize<'a>,
```

This is equivalent to `serde`'s `DeserializeOwned`. You can then remove all the cloning below and the intermediate value.